effekt/shared/src/main/scala/effekt/machine/Transformer.scala

```diff
@@ -423,7 +433,14 @@ object Transformer {
 
     case symbols.InterfaceType(interface, List()) => Negative(interface.name.name)
 
-    case symbols.ValueTypeApp(typeConstructor, List()) => Positive(typeConstructor.name.name)
+    case symbols.ValueTypeApp(typeConstructor, targs) =>
+      // TODO: Add something like:
+      // if(!targs.forall(isSimpleDataType)) {
+      //   Context.abort(s"Unsupported type ${tpe} polymorphic over non-data type")
+      // }
```

Comment:
Can you try to just translate the `targs` here and automatically fail?

Reply:
BTW if you don't add this check now, then we probably never will ;)

Reply:
Yes, it's probably not a big problem (PolymorphismBoxing should have removed all of them). Actually, IIUC, the whole function should ideally be dropped (it is `transform(symbols.Type)` which we don't want to use anymore, but it is still called from https://github.com/effekt-lang/effekt/pull/199/files/f50fb68cac88083e264fd21b96cfbaf0a61c89d4#diff-2bc35700065c09eb82c800736c2625a2516ddb669bb20acbfce3d9af8dbca683R343 for the types of fields.
I just noticed that this is now annotated with a type anyway, so we can use that and drop this function.

Reply:
That function is now gone.